### PR TITLE
PM & PMD Create/Edit

### PR DESCRIPTION
**Pull request by scott062**
https://github.com/MetaPhase-Consulting/mock-fsbid/pull/413
https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2756

Need to update and confirm HRTMAP responses before merging

### CODE REVIEW

File: talentmap_api/fsbid/services/admin_panel.py [hunk 1]
```diff
@@ -86,46 +88,177 @@ def success_mapping(x):
...
+    # Check for existing panel meeting
+    existing_pm_seq_num = ref.get("pm_seq_num")
+    # Change back to none
+    newly_created_pm_seq_num = 1
+
+    if panel_status or panel_type:
+        if not existing_pm_seq_num:
+            panel_meeting = create_panel_meeting(query, jwt_token)
+            # TO DO: verify res for edit and create consistency
+            # newly_created_pm_seq_num = pydash.get(panel_meeting, "[0].pm_seq_num", None)
```
Comment (scott062) [inline talentmap_api/fsbid/services/admin_panel.py:120]: Will update before merging

File: talentmap_api/fsbid/services/admin_panel.py [hunk 2]
```diff
@@ -86,46 +88,177 @@ def success_mapping(x):
...
+    cut_date = query.get("prelimCutoff")
+
+    # Unpack original resource 
+    ref = query.get("originalReference", {})
+    refDates = ref.get("panelMeetingDates", {})
+
+    # Check for existing panel meeting
+    existing_pm_seq_num = ref.get("pm_seq_num")
+    # Change back to none
+    newly_created_pm_seq_num = 1
```
Comment (scott062) [inline talentmap_api/fsbid/services/admin_panel.py:114]: Will update before merging

File: talentmap_api/fsbid/services/common.py [hunk 3]
```diff
@@ -321,10 +321,12 @@ def send_get_request(uri, query, query_mapping_function, jwt_token, mapping_func
 
 def send_put_request(uri, query, query_mapping_function, jwt_token, mapping_function, api_root=API_ROOT):
     mappedQuery = pydash.omit_by(query_mapping_function(query), lambda o: o is None)
+    print("starting put request")
+    logger.info(uri)
     logger.info(mappedQuery)
     url = f"{api_root}/{uri}"
     logger.info(url)
-    response = requests.post(url, headers={'JWTAuthorization': jwt_token, 'Content-Type': 'application/json'}, json=mappedQuery).json()
+    response = requests.put(url, headers={'JWTAuthorization': jwt_token, 'Content-Type': 'application/json'}, json=mappedQuery).json()
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/common.py:329]: Is this gonna be an issue for remarks? it's the only other place that uses this function, but idk if we actually use that remarks view and service
image
image

File: talentmap_api/fsbid/views/admin_panel.py [hunk 4]
```diff
@@ -65,6 +65,16 @@ class FSBidPanelMeetingActionView(APIView):
         }
     ))
 
+    def post(self, request):
+        '''
+        Create Panel Meeting
+        '''
+        result = services.modify_panel_meeting_and_dates(request.data, request.META['HTTP_JWT'])
+        # if result is None:
+        #    return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/views/admin_panel.py:74]: 🥶

File: talentmap_api/fsbid/services/panel.py [hunk 5]
```diff
@@ -24,6 +24,10 @@
     'pmdcreatedate': 'pmd_create_date',
     'pmdcreateid': 'pmd_create_id',
     'pmdmdtcode': 'mdt_code',
+    'pmdupdatedate': 'pmd_update_date',
+    'pmdupdateid': 'pmd_update_id',
+    'pmdcreatedate': 'pmd_create_date',
+    'pmdcreateid': 'pmd_create_id',
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/panel.py:30]: These were already here on lines 22-25

File: talentmap_api/fsbid/services/admin_panel.py [hunk 6]
```diff
@@ -86,46 +88,177 @@ def success_mapping(x):
...
     }
+    return mapped_query
+
+
+
+# ======================== Edit Panel Meeting ========================
+def edit_panel_meeting(query, jwt_token):
+    pm_seq_num = query.get("originalReference", {}).get("pm_seq_num")
+    args = {
+        "uri": f"v1/panels/meeting/{pm_seq_num}",
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/admin_panel.py:221]:
```suggestion
    pmseqnum = query.get("originalReference", {}).get("pm_seq_num")
    args = {
        "uri": f"v1/panels/meeting/{pmseqnum}",
```

just for consistency

File: talentmap_api/fsbid/services/admin_panel.py (hunk 2 above)
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/admin_panel.py:114]: don't forget to change and also L120

**Comment by elizabeth-jimenez** [comment]
just a few small changes needed and just a reminder to revert those lines you set up for locally

File: talentmap_api/fsbid/services/panel.py (hunk 5 above)
Comment (scott062) [reply talentmap_api/fsbid/services/panel.py:30]: yea i think i hadn't merged that branch at the time

**Comment by scott062** [comment]
All feedback should be addressed

**Pull request merged**